idaes/models_extra/gas_distribution/parsers/tests/test_networkdata_model.py

```diff
@@ -0,0 +1,140 @@
+    assert(len(n1.inlets) == 1 and len(n1.outlets) == 2)
+    assert(len(n2.inlets) == 1 and len(n2.outlets) == 2)
+    assert(len(n3.inlets) == 2 and len(n3.outlets) == 1)
+    assert(len(n4.inlets) == 2 and len(n4.outlets) == 1)
+    assert(len(n5.inlets) == 1 and len(n5.outlets) == 2)
+    assert(len(x1.inlets) == 1 and len(x1.outlets) == 0)
+    assert(len(x2.inlets) == 1 and len(x2.outlets) == 0)
+    assert(len(x3.inlets) == 1 and len(x3.outlets) == 0)
+
+    # e1_e3 connection
+    e1_e3 = m.fs.pipelines['pipe01_entry01_entry03']
+    assert(e1.outlets[0].arc.ports[0] == e1_e3.inlet_port)
+    assert(e3.inlets[0].arc.ports[0] == e1_e3.outlet_port)
+    assert (e1.outlets[0].arc.ports[1] == e1.outlets[0].port and e3.inlets[0].arc.ports[1] == e3.inlets[0].port)
+
+    # e3_n1 connection - compressor
+    e3_n1 = m.fs.compressors['CS01_entry03_N01']
+    assert(e3.outlets[0].arc.ports[0] == e3_n1.inlet_port)
+    assert(n1.inlets[0].arc.ports[0] == e3_n1.outlet_port)
+    assert(e3.outlets[0].arc.ports[1] == e3.outlets[0].port and n1.inlets[0].arc.ports[1] == n1.inlets[0].port)
+
+
+    # n1_n2 connection - pipe
+    n1_n2 = m.fs.pipelines['pipe02_N01_N02']
+    assert (n1.outlets[0].arc.ports[0] == n1_n2.inlet_port)
+    assert (n2.inlets[0].arc.ports[0] == n1_n2.outlet_port)
+
+    # n1_n3 connection - direct connection
+    assert(n1.outlets[1].arc.ports[0] == n1.outlets[1].port)
+    assert(n1.outlets[1].arc.ports[1] == n3.inlets[1].port)
+
+    # e2_n3 connection - pipe
+    e2_n3 = m.fs.pipelines['pipe03_entry02_N03']
+    assert(e2.outlets[0].arc.ports[0] == e2_n3.inlet_port)
+    assert(n3.inlets[0].arc.ports[0] == e2_n3.outlet_port)
+
+    # n2_x1 connection - pipe
+    n2_x1 = m.fs.pipelines['pipe04_N02_exit01']
+    assert(n2.outlets[0].arc.ports[0] == n2_x1.inlet_port)
+    assert(x1.inlets[0].arc.ports[0] == n2_x1.outlet_port)
+
+    # n2_n4 connection - pipe
+    n2_n4 = m.fs.pipelines['pipe05_N02_N04']
+    assert(n2.outlets[1].arc.ports[0] == n2_n4.inlet_port)
+    assert(n4.inlets[0].arc.ports[0] == n2_n4.outlet_port)
+
+    # n3_n4 connection - pipe
+    n3_n4 = m.fs.pipelines['pipe06_N03_N04']
+    assert(n3.outlets[0].arc.ports[0] == n3_n4.inlet_port)
+    assert(n4.inlets[1].arc.ports[0] == n3_n4.outlet_port)
+
+    # n4_n5 connection - compressor
+    n4_n5 = m.fs.compressors['CS02_N04_N05']
+    assert(n4.outlets[0].arc.ports[0] == n4_n5.inlet_port)
+    assert(n5.inlets[0].arc.ports[0] == n4_n5.outlet_port)
+
+    # n5_x2 connection - pipe
+    n5_x2 = m.fs.pipelines['pipe07_N05_exit02']
+    assert(n5.outlets[0].arc.ports[0] == n5_x2.inlet_port)
+    assert(x2.inlets[0].arc.ports[0] == n5_x2.outlet_port)
+
+    # n5_x3 connection - pipe
+    n5_x3 = m.fs.pipelines['pipe08_N05_exit03']
+    assert(n5.outlets[1].arc.ports[0] == n5_x3.inlet_port)
+    assert(x3.inlets[0].arc.ports[0] == n5_x3.outlet_port)
+
+@pytest.mark.unit
+def test_nd_plot():
+    net.plot_results_at_time(m, m.fs.time.first(), show_plot=False)
+
+
```

Comment:
Can you add a test that checks that the model has zero degrees of freedom? Maybe something like:
```python
from pyomo.contrib.incidence_analysis import IncidenceGraphInterface

def test_model_zero_dof():
    for node in m.fs.nodes.values():
        for supply in node.supplies.values():
            supply.state[:].pressure.fix()
    for comp in m.fs.compressors.values():
        comp.boost_pressure[:].fix()
    igraph = IncidenceGraphInterface(m, include_inequality=False)
    var_dmp, con_dmp = igraph.dulmage_mendelsohn()
    assert not var_dmp.unmatched and not con_dmp.unmatched
```
Alternatively you could use `assert not degrees_of_freedom(m)` where `degrees_of_freedom` comes from `idaes.core.util.model_statistics`, but I like Dulmage-Mendelsohn because this also makes sure the model is structurally nonsingular.